ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/ContractKeysValidator.scala

```diff
@@ -110,19 +113,19 @@ private[transaction] object ContractKeysValidation {
           case Inconsistent =>
             "InconsistentKeys: at least one contract key has changed since the submission"
         }
-        transactionCommitter.reject(
-          recordTime,
-          transactionCommitter.buildRejectionLogEntry(
+        transactionRejector.reject(
```

Comment:
I'd recommend further refactoring how transaction rejections are built and adding a separate method to `TransactionRejector` for constructing a `StepStop` with a specific rejection entry (i.e., encapsulate more of the logic within the latter class so that callers don't have to invoke two separate functions `buildRejectionEntry` and `reject`). It's perfectly fine if we do the latter refactoring in a separate PR to keep this one easier to follow.

Reply:
addressed in: https://github.com/digital-asset/daml/pull/10258